.github/workflows/validate.yaml

```diff
@@ -0,0 +1,55 @@
+name: Validate Definitions
+
+on:
+  push:
+    branches:
+      - main   # push commit to the main branch
+  pull_request:
+    branches:
+      - main   # pull request to the main branch
+
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-linux:
+    name: CI py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip setuptools
+          python -m pip install -r requirements.txt
+
+      - name: validate
+        run: |
+          python -m dev_tools nxtest
+
+      - name: cnxvalidate build deps
+        run: |
+          sudo apt-get update -y && \
+          sudo apt-get install -y \
+             build-essential libhdf5-dev
+
+      - name: install cnxvalidate
```

Comment:
Not appropriate for the definitions repository.  This belongs in the cnxvalidate repo.

Reply:
This is step 1 of a plan.